### ISSUE TITLE
Experiment.log_histogram_3d with labels (or create  bar_3d?)

### ISSUE THREAD

**Issue by vict0rsch** [comet-ml/issue-tracking#380]
Would it  be possible to have a kind of logging  (be it an extension of histogram_3d or a new one like a bar plot) allowing for:

* histogram_3d's capacity of illustrating a change of distribution over the course of training
* naming/labeling the x axis

The use-case is that I have a model with very many losses, which I log independently and as a weighted sum, but I'd like to also have a view of the distribution in this weighted sum.

Toy example:

```
losses = {}
losses["segmentation"] = seg_loss * lambdas["segmentation"]
losses["depth"] = depth_loss * lambdas["depth"]
losses["domain_adaptation"] = da_loss * lambdas["domain_adaptation"]
losses["total_variational"] = tv_loss * lambdas["total_variational"]
losses["discriminator"] = disc_loss * lambdas["discriminator"]
...

loss = sum(losses.values())
loss.backward()

exp.log_thinggy("loss distribtution", x=losses.keys(), y=[l.item() for l in losses.values()], step=12)
```

Thanks for your time :)

**Comment by dsblank** [comment]
@vict0rsch Thanks for the suggestion!

Have you seen the new [Comet Panels](https://www.comet.ml/docs/user-interface/gallery/)? This makes it fairly easy to log your own data structures, and then visualize them however you wish. 

In fact, there are a couple of Panels from the Gallery that may be close to what you would like to do, including:

* https://www.comet.ml/demo/gallery/view/new#select-panel?gallery-tab=Public&preview-template-id=qhizmWAIlIuuCrrOmV4Om6PUY - shows a way of manually processing histograms
* https://www.comet.ml/demo/gallery/view/new#select-panel?gallery-tab=Public&preview-template-id=iYymacTAePCdFkhyOSnAMAgKr - shows a way of processing custom JSON assets

**Comment by vict0rsch** [comment]
I'm not sure if these examples really apply to what I mean (or maybe I don't understand  them well enough), I really just mean standard bar plots with labels for x, in a 3D view (like `histogram_3d`) to compare the shift in values over the course of training of the different components of the loss

**Comment by dsblank** [comment]
Indeed, neither of the provided examples do exactly what you want, but were meant to give some examples of similar visualizations. The first uses exactly the same data as the built-in Histogram tab 3D chart that you mentioned, with time running left-to-right. The second shows an example of a custom JSON: how to open the asset, read in the data, and display it.

If you could clearly define a JSON data structure (with an example of values) and sketch out some-what graphically what it would look like, we can perhaps help you put together the Panel.

Here are some of the Bar Charts provided by Plotly that might be useful:

* https://plotly.com/javascript/ribbon-plots/
* https://plotly.com/javascript/bar-charts/

I think you could struct something very similar to the Comet 3D Histogram chart with filled-area plots:

* https://plotly.com/javascript/filled-area-plots/

And of course, you can use any JavaScript plotting library, including:

* https://vega.github.io/vega/examples/ (see [here](https://www.comet.ml/demo/gallery/view/new#select-panel?gallery-tab=Public&preview-template-id=JmNeQa5bWa3kfIXKxq6uiiGVl) for a Comet Panel example)
* https://www.chartjs.org/
* https://d3js.org/

**Comment by dsblank** [comment]
It looks like this request is specific to a particular use-case. Which makes it a perfect candidate for a custom Panel.
